crates/driver/src/tests/cases/multiple_solutions.rs

```diff
@@ -0,0 +1,65 @@
+use crate::tests::{
+    setup,
+    setup::new::{Order, Solution},
+};
+
+/// Test that the best-scoring solution is picked when the /solve endpoint
+/// returns multiple valid solutions.
+#[tokio::test]
+#[ignore]
+async fn valid() {
+    let test = setup()
+        .pool(
+            "A",
+            1000000000000000000000u128.into(),
+            "B",
+            600000000000u64.into(),
+        )
+        .order(Order {
+            amount: 500000000000000000u64.into(),
+            sell_token: "A",
+            buy_token: "B",
+            ..Default::default()
+        })
+        .solution(Solution::Valid)
+        // Additional calldata increases the TX size, so the fees are higher resulting in a lower
+        // score.
+        .solution(Solution::AdditionalCalldata { bytes: 5 })
+        .done()
+        .await;
+
+    let solve = test.solve().await;
+
+    solve.ok().score(-46008923437586.0);
```

Comment:
Not related to this PR but it's not really possible to tell by reading the test that the order without the additional bytes is picked or that they actually differ in their score.